lib/livebook_web/live/session_live/persistence_live.ex

```diff
@@ -48,144 +45,79 @@ defmodule LivebookWeb.SessionLive.PersistenceLive do
-                    tabindex="-1"
-                  >
-                    Choose
-                  </button>
-                </div>
-              </.live_component>
-            </div>
-            <div class="mt-6 text-gray-500 text-sm">
-              File: <%= normalize_file(@draft_file).path %>
+
+        <form
+          phx-change="set_options"
+          onsubmit="return false;"
+          class="flex flex-col space-y-4 items-start max-w-full"
+        >
+          <div class="flex flex-col space-y-4">
+            <.switch_checkbox
+              name="persist_outputs"
+              label="Persist outputs"
+              checked={@new_attrs.persist_outputs}
+            />
+            <div class="flex space-x-2 items-center">
+              <span class="text-gray-700 whitespace-nowrap">Autosave</span>
+              <.select
+                name="autosave_interval_s"
+                selected={@new_attrs.autosave_interval_s}
+                options={
+                  [
+                    {5, "every 5 seconds"},
+                    {30, "every 30 seconds"},
+                    {60, "every minute"},
+                    {600, "every 10 minutes"},
+                    {nil, "never"}
+                  ]
+                }
+              />
             </div>
           </div>
-        <% end %>
-        <div class="flex">
-          <%= if @new_attrs.file do %>
-            <button
-              class="button-base button-blue"
-              phx-click="save"
-              disabled={
-                not savable?(@new_attrs, @attrs, @running_files, @draft_file) or
-                  (same_attrs?(@new_attrs, @attrs) and @saved_file == @new_attrs.file)
-              }
-            >
-              Save now
-            </button>
-          <% else %>
-            <button
-              class="button-base button-blue"
-              phx-click="save"
-              disabled={
-                not savable?(@new_attrs, @attrs, @running_files, @draft_file) or
-                  same_attrs?(@new_attrs, @attrs)
-              }
-            >
-              Apply
-            </button>
-          <% end %>
+
+          <span class="text-gray-700 whitespace-nowrap pt-2">
+            File: <%= normalize_file(@draft_file).path %>
+          </span>
+        </form>
+      </div>
+      <div class="flex justify-between">
+        <div class="flex space-x-3">
+          <button
+            class="button-base button-blue"
+            phx-click="save"
+            disabled={not savable?(@draft_file, @saved_file, @running_files)}
+          >
+            Save
+          </button>
+
+          <%= live_patch("Cancel",
+            to: Routes.session_path(@socket, :page, @session.id),
+            class: "button-base button-outlined-gray"
+          ) %>
         </div>
+
+        <%= if @saved_file do %>
+          <button class="button-base button-outlined-red" phx-click="stop_saving">
+            Stop saving
```

Comment:
This could be `Disable autosave`, `Disable file persistence` or `Detach file from session` (sounds more complex than it should)